mu/modes/esp.py

```diff
@@ -0,0 +1,230 @@
+"""
+A mode for working with ESP8266 and ESP32 boards running MicroPython.
+
+Copyright (c) 2015-2019 Nicholas H.Tollervey and others (see the AUTHORS file).
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import logging
+from mu.modes.base import MicroPythonMode, FileManager
+from mu.modes.api import ESP_APIS, SHARED_APIS
+from mu.interface.panes import CHARTS
+from PyQt5.QtCore import QThread
+
+
+logger = logging.getLogger(__name__)
+
+
+class ESPMode(MicroPythonMode):
+    """
+    Represents the functionality required for running MicroPython on ESP8266
+    """
+    name = _('ESP MicroPython')
+    description = _("Write MicroPython on ESP8266/ESP32 boards.")
+    icon = 'esp'
+    fs = None
+
+    valid_boards = [
+        (0x1A86, 0x7523),  # Wemos D1 Mini USB VID, PID
+        (0x10C4, 0xEA60),  # Wemos D1 Mini Pro USB VID, PID
+        (0x0403, 0x6015),   # Sparkfun ESP32 Thing VID, PID
+    ]
```

Comment:
Those are the boards that I have immediately available at hand. Note that about half of the actually use ESP8266.
```
| VID | PID | Chip | Board |
| 10c4 | ea60 | CP210x | Adafruit Feather HUZZAH |
| 10c4 | ea60 | CP210x | AMICA NodeMCU |
| 10c4 | ea60 | CP210x | TTGO-W8266 |
| 10c4 | ea60 | CP210x | Heltec HTIT-W8266 |
| 10c4 | ea60 | CP210x | WeMos Lilin32 |
| 10c4 | ea60 | CP210x | TTGO LORA32 |

| 1a86 | 7523 | HL-340 | Lolin32 LITE |
| 1a86 | 7523 | HL-340 | WeMos D1 Mini LITE |
| 1a86 | 7523 | HL-340 | DFRobot WiFi D1 Mini |
| 1a86 | 7523 | HL-340 | TTGO Mini 32 |
| 1a86 | 7523 | HL-340 | Arduino Nano |
```

Reply:
@deshipu many many thanks. I think your list proves that (at least when it comes to ESP8266 boards) we have the two most common VID/PID combos already covered. ;-)

Reply:
I'll annotate the "metadata" you include into the source code for future reference.

Reply:
(0x0403, 0x6015) is an FTDI FT230X, which is also on the https://www.tindie.com/products/onehorse/esp8285-development-board/ . All other ESP boards I have are covered by the CP210x and HL-340.

Reply:
@cwalther thanks for the heads up! It feels like we're covered as much as one can reasonably expect at this moment in time. Let's see who shouts during the alpha releases, if at all, in this respect. :+1: